File: docs/examples/rs/examples/data_replication_browse.rs
use reduct_rs::{Labels, ReductClient, ReductError};
use tokio;

#[tokio::main]
async fn main() -> Result<(), ReductError> {
    // Create a client instance
    let client = ReductClient::builder()
        .url("http://127.0.0.1:8383")
        .api_token("my-token")
        .build();

    // List all replications
    for replication in client.list_replications().await? {
        println!("Replication: {}", replication.name);
        println!("Active: {}", replication.is_active);
        println!("Pending records: {}", replication.pending_records);
        println!("Provisioned: {}", replication.is_provisioned);
    }

    // Get all details of a replication
    let replication = client.get_replication("example-replication").await?;
    println!("Replication: {}", replication.info.name);
    println!("Settings: {:?}", replication.settings);
    println!(
        "Failed records (last hour): {}",
        replication.diagnostics.hourly.errored
    );
    println!(
        "Successful records (last hour): {}",
        replication.diagnostics.hourly.ok
    );
    println!(
        "Errors (last hour): {:?}",
        replication.diagnostics.hourly.errors
    );

    Ok(())
}

File: versioned_docs/version-1.14.x/examples/cpp/src/data_replication_browse.cc
#include <reduct/client.h>
#include <iostream>
#include <cassert>

using reduct::IBucket;
using reduct::IClient;
using reduct::Error;

int main() {
    // Create a client instance
    auto client = IClient::Build("http://127.0.0.1:8383", {
            .api_token = "my-token"
    });

    // List all replications
    auto [replications, list_err] = client->GetReplicationList();
    for (auto& replication : replications) {
        std::cout << "Replication: " << replication.name << std::endl;
        std::cout << "Active: " << replication.is_active << std::endl;
        std::cout << "Pending records: " << replication.pending_records << std::endl;
        std::cout << "Provisioned: " << replication.is_provisioned << std::endl;
    }

    // Get all details of a replication
    auto [replication, detail_err] = client->GetReplication("example-replication");
    std::cout << "Replication: " << replication.info.name << std::endl;
    std::cout << "Source Bucket: " << replication.settings.src_bucket << std::endl;
    std::cout << "Failed records (last hour): " << replication.diagnostics.hourly.errored << std::endl;
    std::cout << "Successful records (last hour): " << replication.diagnostics.hourly.ok << std::endl;
}
